model/bit_typespec.yaml

```diff
@@ -24,7 +24,7 @@
   - obj_ref: bit_typespec
     name: bit typespec
     vpi: vpiElemTypespec
-    type: bit_typespec
+    type: ref_obj
```

Comment:
Here it is even worse from a specific bit_typespec we now have a ref_obj to anything.
There should be a ref_typespec whose only choices are typespec classes

Reply:
@hs-apotell , I'm sorry, I'm not sure what was I thinking back then. 
We use ref_obj in too many places now, it is almost like a void* (any).
I think now we should go with https://github.com/chipsalliance/UHDM/pull/984

Reply:
Done. Please keep for other PRs all the changes that are not of the matter at hand.

Reply:
Isn't the point of ref_typespec to be a pointer to an actual typespec object that can be shared by multiple ref_typespec?
Else you clone all typespecs which will have a huge memory impact.

Reply:
We should have a field actual_typespec of type typespec.
We need to have an explicit pointer from ref_typespec to a a typespec